src/character.cpp

```diff
@@ -4046,7 +4046,9 @@ void Character::mod_stored_calories( int ncal, const bool ignore_weariness )
     if( !ignore_weariness ) {
         activity_history.calorie_adjust( ncal );
     }
-    set_stored_calories( stored_calories + ncal );
+    if( ncal < 0 || ncal + stored_calories > 0 ) {
```

Comment:
This overflow proofs it.
```suggestion
    if( ncal < 0 || std::numeric_limits<int>::max() - ncal > stored_calories ) {
```